File: server/FitMate.Core/JsonModels/Workouts/WorkoutExerciseGroupModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Workouts;

public class WorkoutExerciseGroupModel
{
    public long Id { get; set; }
    public int SortOrder { get; set; }
    public ExerciseGroupType GroupType { get; set; }
    public List<WorkoutExerciseModel> Exercises { get; set; } = [];
}

File: client/src/types/JsonModels/Workouts/WorkoutExerciseGroupModel.ts
import type { JsonModels } from "../../backend";
import type { ExerciseGroupType } from "../Enums/ExerciseGroupType";

export interface WorkoutExerciseGroupModel
{
	id: number;
	sortOrder: number;
	groupType: ExerciseGroupType;
	exercises: JsonModels.Workouts.WorkoutExerciseModel[];
}
